_posts/en/newsletters/2022-05-04-newsletter.md

```diff
@@ -0,0 +1,188 @@
+  [posted][teinturier security] to the Lightning-Dev mailing list the
+  announcement of a security issue he had previously responsibly
+  disclosed to LN implementation maintainers.  The issue affected older
+  versions of Core Lightning (with experimental features enabled) and
+  LND.
+
+    Prior to the implementation of [anchor outputs][topic anchor
+    outputs], revoked [HTLC][topic HTLC] transactions only contained a
+    single output, so many implementations only tried to claim that
+    single output.  The design of anchor outputs for LN allows combining
+    multiple revoked HTLC outputs into a single transaction, but this
+    is only safe if implementations claim all of the relevant outputs in
+    the transaction.  Teinturier's implementation of anchor outputs for
+    Eclair allowed him to test the other LN implementations and discover
+    the vulnerability.
+
+    As with a previous attack related to anchor outputs (see [Newsletter
+    #115][news115 fee stealing]), the problem appears to be related to
+    adding support for signing with
+    `SIGHASH_SINGLE|SIGHASH_ANYONECANPAY` while still retaining legacy
+    support for signing with `SIGHASH_ALL`.
+
+- **LN gossip rate limiting:** Alex Myers [posted][myers recon] to the
+  Lightning-Dev mailing list about his research into using
+  [minisketch][topic minisketch]-based set reconciliation to reduce the
+  amount of bandwidth nodes need to learn about updates to the LN
+  channel graph.  His method assumes all peers have almost all of the
+  same information about almost all of the same public channels.  One
+  peer can then generate a minisketch from its complete graph of the
+  public network and send that to all of its peers, who can use the
+  minisketch to find any updates to the network since its last
+  reconciliation.  This is different than the proposed use of minisketch
+  for the Bitcoin P2P network via the [erlay][topic erlay] protocol
+  where only updates (new unconfirmed transactions) from the last few
+  seconds are sent.
+
+    One challenge of reconciling over all public channels is that it
+    requires all LN nodes keep the same information.  Any filtering that
+    produces a persistent difference in the view of the channel graph
+    between nodes will result in either a bandwidth overhead or a
+    failure of the protocol.  Matt Corallo [suggested][corallo recon]
+    that this could be addressed by applying the erlay model to LN---if
+    only new information was synced, there wouldn't be a concern about
+    persistent differences, although large variations in filtering rules
+    could still result in wasted bandwidth or reconciliation failure.
+    Myers was concerned about the amount of state tracking required by
+    only sending updates---a Bitcoin Core node maintains separate state
+    for each of its peers in order to avoiding resending them updates
+    they previously sent to the node.  The alternative of reconciling
```

Comment:
```suggestion
    only sending updates---a Bitcoin Core node maintains a separate state
    for each of its peers in order to avoid resending updates
    they previously sent to the node.  The alternative of reconciling
```